diff --git a/src/Makefile.am b/src/Makefile.am
index d2265aa..a492c5b 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -21,8 +21,8 @@ prelink_SOURCES = cache.c conflict.c cxx.c doit.c exec.c execle_open.c get.c \
 		  verify.c canonicalize.c md5.c md5.h sha.c sha.h 	     \
 		  $(common_SOURCES) $(arch_SOURCES)
 prelink_LDADD = @LIBGELF@
-prelink_LDFLAGS = -all-static
 
 execstack_SOURCES = execstack.c $(common_SOURCES) $(arch_SOURCES)
+execstack_LDADD = @LIBGELF@
 
 extra_DIST = makecrc.c
diff --git a/src/Makefile.in b/src/Makefile.in
index 85cf845..aa2c2a6 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -119,7 +119,6 @@ prelink_SOURCES = cache.c conflict.c cxx.c doit.c exec.c execle_open.c get.c \
 		  $(common_SOURCES) $(arch_SOURCES)
 
 prelink_LDADD = @LIBGELF@
-prelink_LDFLAGS = -all-static
 
 execstack_SOURCES = execstack.c $(common_SOURCES) $(arch_SOURCES)
 
@@ -158,7 +157,7 @@ prelink_DEPENDENCIES =
 DEFAULT_INCLUDES =  -I. -I$(srcdir) -I$(top_builddir)
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lgelf -L$(top_srcdir)/gelf
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 @AMDEP_TRUE@DEP_FILES = ./$(DEPDIR)/arch-alpha.Po \
