diff --git a/src/arch-arm.c b/src/arch-arm.c
index 30a8259..e7a490f 100644
--- a/src/arch-arm.c
+++ b/src/arch-arm.c
@@ -832,7 +832,7 @@ PL_ARCH = {
   .R_COPY = R_ARM_COPY,
   .R_RELATIVE = R_ARM_RELATIVE,
   .rtype_class_valid = RTYPE_CLASS_VALID,
-  .dynamic_linker = "/lib/ld-linux.so.3",
+  .dynamic_linker = "/lib/ld-linux-armhf.so.3",
   .adjust_dyn = arm_adjust_dyn,
   .adjust_rel = arm_adjust_rel,
   .adjust_rela = arm_adjust_rela,
