diff -Naur prelink/src/arch-x86_64.c prelink.tpg/src/arch-x86_64.c
--- prelink/src/arch-x86_64.c	2009-11-03 15:45:10.000000000 +0000
+++ prelink.tpg/src/arch-x86_64.c	2013-01-27 02:22:00.323375557 +0000
@@ -532,7 +532,7 @@
   .R_COPY = R_X86_64_COPY,
   .R_RELATIVE = R_X86_64_RELATIVE,
   .rtype_class_valid = RTYPE_CLASS_VALID,
-  .dynamic_linker = "/lib64/ld-linux-x86-64.so.2",
+  .dynamic_linker = "/lib/ld-linux-x86-64.so.2",
   .adjust_dyn = x86_64_adjust_dyn,
   .adjust_rel = x86_64_adjust_rel,
   .adjust_rela = x86_64_adjust_rela,
